core/src/main/java/org/elasticsearch/indices/query/IndicesQueriesRegistry.java

```diff
@@ -19,24 +19,47 @@
 
 package org.elasticsearch.indices.query;
 
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.ParsingException;
 import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentLocation;
 import org.elasticsearch.index.query.QueryParser;
 
 import java.util.Map;
 
 public class IndicesQueriesRegistry extends AbstractComponent {
     private Map<String, QueryParser<?>> queryParsers;
+    private Map<String, ParseField> parseFields;
+    private ParseFieldMatcher parseFieldMatcher;
 
-    public IndicesQueriesRegistry(Settings settings, Map<String, QueryParser<?>> queryParsers) {
+    public IndicesQueriesRegistry(Settings settings, Map<String, QueryParser<?>> queryParsers, Map<String, ParseField> parseFields) {
```

Comment:
Sorry I understood only later what Nik meant, I am changing to Tuple instead of two maps. thanks for pointing that out. I thought we were talking about changing the fromXContent method of the functional interface to something that would use Tuple... but using Tuple internally is fine and actually saves me from some synchronization issues that need to be handled is we have two maps.